src/formatter.js

```diff
@@ -452,11 +471,14 @@ let _getFileDetails = function (logEvent) {
 
 		logEvent.column = fileParts.pop();
 		logEvent.lineNumber = fileParts.pop();
-
+		
 		if (typeof define !== 'undefined') {
 			let path = require('path');
 			let appDir = path.dirname(require.main.filename);
-			logEvent.filename = fileParts.join(':').replace(appDir, '').replace(/(\\|\/)/, '');
+			if (!fileParts[0].startsWith(appDir)) {
+				appDir = '';
+			}
+			logEvent.filename = fileParts.join(':').replace(appDir, '').replace(/^(\\|\/)/, '');
```

Comment:
I got situations where appDir was "." or "/" while that wasn't the beginning of fileParts[0], resulting in the first "/" or "." later on in the string being removed. This solution should prevent this.